src/lib.rs

```diff
@@ -3471,3 +3495,13 @@ fn which(tool: &Path) -> Option<PathBuf> {
         return if check_exe(&mut exe) { Some(exe) } else { None };
     })
 }
+
+/// Check if the file's extension is either "asm" or "s", case insensitive.
+fn is_asm(file: &Path) -> bool {
+    if let Some(ext) = file.extension() {
+        let ext = ext.to_ascii_lowercase();
+        ext == "asm" || ext == "s"
```

Comment:
This probably will need some adjustment over time, but I don't see anything it's concretely missing now.